.clang-format

```diff
@@ -189,3 +189,6 @@ WhitespaceSensitiveMacros:
   - NS_SWIFT_NAME
   - CF_SWIFT_NAME
 ...
+---
+Language: Json
+BasedOnStyle: LLVM
```

Comment:
I should have better explained why this is here.

Adding CMake introduces `CMakePresets.json`. Without this change CI will fail because `clang-format` will try to format json file without knowing how. The alternative change I could make here is to make `clang-format` not format JSON files. I'm nots sure how but I'm sure there's a way.

Reply:
I can separate this to a separate commit explaining why tho.